third_party/abseil-cpp/absl/log/internal/log_sink_set.cc

```diff
@@ -58,8 +58,12 @@ bool& ThreadIsLoggingStatus() {
 #ifdef ABSL_HAVE_THREAD_LOCAL
   ABSL_CONST_INIT thread_local bool thread_is_logging = false;
   return thread_is_logging;
+#else
+#if defined(STARBOARD)
+  static pthread_key_t thread_is_logging_key;
 #else
   ABSL_CONST_INIT static pthread_key_t thread_is_logging_key;
```

Comment:
I think this means that the `pthread_key_t` that evergreen compiles with does not have a constant initializer (see the definition of `ABSL_CONST_INIT` in `third_party/abseil-cpp/absl/base/attributes.h`). It may be worthwhile to file a bug with the platforms team to look into this.

Reply:
Added TODO for b/328310825 (fyi @y4vor )